### PR TITLE
Fix/add new list in geo view

### PR DESCRIPTION
**Pull request by AdityaKhatri**
Depends on https://github.com/toggle-corp/re-map/pull/4

### CODE REVIEW

File: src/components/geo/RegionMap/index.js [hunk 1]
```diff
@@ -581,6 +582,7 @@ class RegionMap extends React.PureComponent {
                 ...feature,
                 properties: {
                     ...feature.properties,
+                    color: colors.Paired[12][Math.floor(Math.random() * 12)],
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]: Let's create a function for this one?

File: src/components/geo/RegionMap/index.js [hunk 2]
```diff
@@ -779,15 +779,18 @@ class RegionMap extends React.PureComponent {
                         navControlShown={false}
                     >
                         <MapContainer
-                            className={styles.geoJsonMap}
+                            className={_cs(
+                                styles.geoJsonMap,
+                                !editMode && styles.disableMapDraw,
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]: We should move this inside the MapShapeEditor

File: src/components/input/GeoInput/GeoModal/GeoInputList/index.js [hunk 3]
```diff
@@ -235,17 +300,40 @@ const GeoInputList = (props) => {
...
-        [handlePolygonEdit, handlePolygonRemove, polygonDisabled],
+        [handlePolygonEdit, handlePolygonRemove, polygonDisabled, onPolygonsChange, readOnly],
     );
 
     const newSelections = useMemo(
-        () => createCombinedSelections(selections, polygons),
-        [selections, polygons],
+        () => {
+            const tempSelections = [...createCombinedSelections(selections, polygons)]
+                .sort((a, b) => compareString(
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/GeoInputList/index.js]: Doesn't make sense to sort before sort!

File: src/components/input/GeoInput/GeoModal/GeoInputList/index.js [hunk 4]
```diff
@@ -235,17 +300,40 @@ const GeoInputList = (props) => {
...
+
+            if (sortType === 'alphabetical') {
+                return tempSelections.sort((a, b) => compareString(
+                    geoOptionsById[a.id].title,
+                    geoOptionsById[b.id].title,
+                    sortOrderValue,
+                ));
+            } else if (sortType === 'overlaps') {
+                return tempSelections.sort((a, b) => compareNumber(
+                    (a.polygons && a.polygons.length) || -1,
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/GeoInputList/index.js:329]: a.

Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/GeoInputList/index.js]:
```suggestion
                    (a.polygons ? a.polygons.length : 0),
```

File: src/components/input/GeoInput/GeoModal/GeoInputList/index.js [hunk 5]
```diff
@@ -235,17 +300,40 @@ const GeoInputList = (props) => {
...
+                    (b.polygons && b.polygons.length) || -1,
+                    sortOrderValue,
```
Comment (tnagorra) [inline src/components/input/GeoInput/GeoModal/GeoInputList/index.js]: We should have a fallback compare strategy when they have the same length of polygons.
Maybe serialize the polygon ids?
And for same polygon overlaps, have fallback for alphabetical or something

Comment (AdityaKhatri) [reply src/components/input/GeoInput/GeoModal/GeoInputList/index.js]: Yeah, I sorted the data before making it the fallback.
So, if they have the same number of polygons, they'll be sorted alphabetically.

Comment (AdityaKhatri) [reply src/components/input/GeoInput/GeoModal/GeoInputList/index.js]: Done

File: src/components/geo/RegionMap/index.js [hunk 6]
```diff
@@ -67,6 +67,9 @@ const polygonSourceOptions = {
     promoteId: 'code',
 };
 
+// Paired returns quantitative colors from colorbrewer
+const getRandomColor = () => colors.Paired[12][Math.floor(Math.random() * 12)];
```
Comment (tnagorra) [inline src/components/geo/RegionMap/index.js]:
```suggestion
const getRandomColor = () => getRandomFromList(colors.Paired[12]);
```

getRandomFromList is in fujs

**Pull request merged**